syncapi/storage/sqlite3/deltas/2022061412000000_history_visibility_column.go

```diff
@@ -51,6 +54,41 @@ func UpAddHistoryVisibilityColumnCurrentRoomState(ctx context.Context, tx *sql.T
 	if err != nil {
 		return fmt.Errorf("failed to execute upgrade: %w", err)
 	}
+
+	// get the current room history visibility
+	rows, err := tx.Query(`SELECT DISTINCT room_id, headered_event_json FROM syncapi_current_room_state
+		WHERE type = 'm.room.history_visibility' AND state_key = '';
+`)
+	if err != nil {
+		return fmt.Errorf("failed to query current room state: %w", err)
+	}
+	var eventBytes []byte
+	var roomID string
+	var event gomatrixserverlib.HeaderedEvent
+	var hisVis gomatrixserverlib.HistoryVisibility
+	historyVisibilities := make(map[string]gomatrixserverlib.HistoryVisibility)
+	for rows.Next() {
+		if err = rows.Scan(&roomID, &eventBytes); err != nil {
+			return fmt.Errorf("failed to scan row: %w", err)
+		}
+		if err = json.Unmarshal(eventBytes, &event); err != nil {
+			return fmt.Errorf("failed to unmarshal event: %w", err)
+		}
+		historyVisibilities[roomID] = gomatrixserverlib.HistoryVisibilityShared
+		if hisVis, err = event.HistoryVisibility(); err == nil {
+			historyVisibilities[roomID] = hisVis
+		}
+	}
+
+	// update the history visibility
+	for roomID, hisVis = range historyVisibilities {
```

Comment:
So this migration is a bit wrong because aiui it will load the _current_ history visibility value then apply this for _all_ events? Probaby good enough, but definitely needs comments.